prog/minio/minio_server_nexus.rb

```diff
@@ -106,8 +106,8 @@ def cluster
 
   label def wait
     when_checkup_set? do
```

Comment:
I'm a bit puzzled by this re-arrangement, and perhaps the arrangement of decr_checkup after the hop_unavailable branch generally. I presume `label unavailable` will require success of `available?` at least once before coming back to `wait`, and then in such a case checkup is still incremented, so there will be one more attempt to confirm health before finally decrementing the semaphore. Is that what you want? Or do you want to unconditionally decrement the semaphore and let the jump to `unavailable` take care of everything, taking advantage of transactions?

Reply:
`unavailable` already decrements the semaphore before moving back to `wait`, so `decr_checkup` here was unnecessary.